### PR TITLE
improvements to task selection page + prepare style to dark mode

### PR DESCRIPTION
**Pull request by willemarcel**

### CODE REVIEW

**Comment by willemarcel** [comment]
@thadk I pushed a new paginator to the taskSelection page. Could you check if the paginator is working well in the projects list page?

File: frontend/src/components/taskSelection/taskList.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    } else {
+      dispatch({ type: 'SET_ACTIVE_TASKS', activeTasks: [data.taskId] });
+    }
+    if (activeProject !== projectId) {
+      dispatch({ type: 'SET_ACTIVE_PROJECT', activeProject: projectId });
+    }
+  }
+
+  return (
+    <div className={`cf db pv3 ba br1 mt2 ${activeTasks.includes(data.taskId) ? 'b--blue-dark bw1' : 'b--tan bw1'}`}>
```
Comment (thadk) [inline frontend/src/components/taskSelection/taskList.js]: These left margins get a bit weird when you make the page smaller.
image

File: frontend/src/components/taskSelection/index.js [hunk 2]
```diff
@@ -1,133 +1,114 @@
...
+  return (
+    <div>
+      <div className="cf pv3">
+        <div className="w-100 w-50-ns fl">
+          <div className="ph4">
+            <ReactPlaceholder
+              showLoadingAnimation={true}
+              rows={3}
+              delay={500}
+              ready={typeof(project.projectId) === 'number' && project.projectId > 0}
```
Comment (thadk) [inline frontend/src/components/taskSelection/index.js]: The best way to do these `ready={}` is to use the `!loading` state that is coming in from `useFetch` – I think I tried it and it wasn't completely reliable though. Not sure why.

File: frontend/src/components/paginator/tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+
+import { listPageOptions, howManyPages, PageButton, PaginatorLine } from '../index'
+
+it('test if listPageOptions returns the correct options', () => {
```
Comment (thadk) [inline frontend/src/components/paginator/tests/index.test.js]: 👍

**Comment by thadk** [comment]
Also for the map, it may help to put the task layer under the label layer as described here: https://docs.mapbox.com/mapbox-gl-js/example/geojson-layer-in-stack/

File: frontend/src/components/taskSelection/index.js [hunk 4]
```diff
@@ -1,133 +1,114 @@
...
-                {this.renderHeaderLine()}
-                <div className="cf pb3">
-                  <h3 className="f2 fw6 mt2 mb3 ttu barlow-condensed blue-dark">
-                    {this.props.project.projectInfo && this.props.project.projectInfo.name}
-                  </h3>
-                  <span className="blue-light">{this.props.project.campaignTag} &#183; Brazil</span>
+export function TaskSelection({project, type, loading}: Object) {
+  const [activeSection, setActiveSection] = useState('tasks');
+  const defaultEditor = useSelector(state => state.preferences.default_editor);
+  const [error, tasksLoading, tasks] = useFetch(`projects/${project.projectId}/tasks/`);
```
Comment (thadk) [inline frontend/src/components/taskSelection/index.js]: ${project.projectId} can become `undefined` like with:
`http://localhost:3000/projects/1999/map`
https://tasks-stage.hotosm.org/projects/1999/map

There must be some other related issue too. You could check on this fetch so it cannot crash in other situations. Here is the localhost log (the order is not expected):
```
app           | 172.25.0.1 - - [04/Oct/2019 04:28:17] "OPTIONS /api/v2/projects/1999/queries/summary/ HTTP/1.1" 200 -
app           | 172.25.0.1 - - [04/Oct/2019 04:28:17] "OPTIONS /api/v2/projects/undefined/tasks/ HTTP/1.1" 200 -
app           | 172.25.0.1 - - [04/Oct/2019 04:28:18] "GET /api/v2/projects/undefined/tasks/ HTTP/1.1" 200 -
app           | [2019-10-04 04:28:18,134] CRITICAL in resources: Project Summary GET - unhandled error: 'NoneType' object is not iterable
app           | 172.25.0.1 - - [04/Oct/2019 04:28:18] "GET /api/v2/projects/1999/queries/summary/ HTTP/1.1" 500 -
```

File: frontend/src/components/taskSelection/index.js [hunk 5]
```diff
@@ -1,133 +1,114 @@
...
 import { Link } from '@reach/router';
 import { FormattedMessage } from 'react-intl';
 import ReactPlaceholder from 'react-placeholder';
 
 import messages from './messages';
-import { TasksMap } from './map.js';
+import { useFetch } from '../../hooks/UseFetch';
 import { PriorityBox } from '../projectcard/projectCard';
+import { TasksMap } from './map.js';
 import { TaskSelectionFooter } from './footer';
```
[the remark this answers is not in the document]
Comment (willemarcel) [reply frontend/src/components/taskSelection/index.js]: Fixed it.

File: frontend/src/components/taskSelection/index.js (hunk 2 above)
Comment (willemarcel) [reply frontend/src/components/taskSelection/index.js]: On that case, the projectId is not coming from the useFetch request. I made the verification that way to solve some lint warnings

File: frontend/src/components/taskSelection/index.js (hunk 4 above)
Comment (willemarcel) [reply frontend/src/components/taskSelection/index.js]: I added the not found content when the projectId doesn't exist. Thanks for noticing it!

File: frontend/src/components/taskSelection/taskList.js (hunk 1 above)
Comment (willemarcel) [reply frontend/src/components/taskSelection/taskList.js]: I'll solve it on a future PR.

**Pull request merged**